File: dsa/patterns/18_strings/python/first_repeating_character.py
# Problem: https://www.geeksforgeeks.org/find-the-first-repeated-character-in-a-string/
def first_recurring_character(input_str):
    seen = set()

    for c in input_str:
        if c in seen:
            return c
        seen.add(c)

    return None   # equivalent to returning 0 char in Java


# ------------------ Test (Same as Java main) ------------------

print(first_recurring_character("DBBCAA"))  # Expected: 'B'

File: dsa/patterns/18_strings/java/FirstRepeatingCharacter.java
import java.util.HashMap;
import java.util.HashSet;
import java.util.Set;

public class FirstRepeatingCharacter {

	public static void main(String[] args) {
		FirstRepeatingCharacter obj = new FirstRepeatingCharacter();

		System.out.println(obj.firstRecurringCharacter("DBBCAA"));
	}

	// Problem: https://www.geeksforgeeks.org/find-the-first-repeated-character-in-a-string/
	public char firstRecurringCharacter(String input){
		Set<Character> set = new HashSet<>();

		for(char c: input.toCharArray()){
			if(set.contains(c))
				return c;
			set.add(c);
		}

		return 0;
	}

}
